sklearn/metrics/cluster/tests/test_unsupervised.py

```diff
@@ -15,31 +17,43 @@
 def test_silhouette():
     # Tests the Silhouette Coefficient.
     dataset = datasets.load_iris()
-    X = dataset.data
+    X_dense = dataset.data
+    X_csr = csr_matrix(X_dense)
+    X_dok = sp.dok_matrix(X_dense)
+    X_lil = sp.lil_matrix(X_dense)
     y = dataset.target
-    D = pairwise_distances(X, metric='euclidean')
-    # Given that the actual labels are used, we can assume that S would be
-    # positive.
-    silhouette = silhouette_score(D, y, metric='precomputed')
-    assert(silhouette > 0)
-    # Test without calculating D
-    silhouette_metric = silhouette_score(X, y, metric='euclidean')
-    assert_almost_equal(silhouette, silhouette_metric)
-    # Test with sampling
-    silhouette = silhouette_score(D, y, metric='precomputed',
-                                  sample_size=int(X.shape[0] / 2),
-                                  random_state=0)
-    silhouette_metric = silhouette_score(X, y, metric='euclidean',
-                                         sample_size=int(X.shape[0] / 2),
-                                         random_state=0)
-    assert(silhouette > 0)
-    assert(silhouette_metric > 0)
-    assert_almost_equal(silhouette_metric, silhouette)
-    # Test with sparse X
-    X_sparse = csr_matrix(X)
-    D = pairwise_distances(X_sparse, metric='euclidean')
-    silhouette = silhouette_score(D, y, metric='precomputed')
-    assert(silhouette > 0)
+
+    for X in [X_dense, X_csr, X_dok, X_lil]:
+        D = pairwise_distances(X, metric='euclidean')
+        # Given that the actual labels are used, we can assume that S would be
+        # positive.
+        score_precomputed = silhouette_score(D, y, metric='precomputed')
+        assert_greater(score_precomputed, 0, "silhouette score is not greater than 0")
```

Comment:
PEP8 line length!

And actually I think the explicit error message adds little. Just use the default.